src/txn/txn_rollback_to_stable.c

```diff
@@ -960,6 +960,82 @@ __rollback_to_stable_btree(WT_SESSION_IMPL *session, wt_timestamp_t rollback_tim
     return (ret);
 }
 
+/*
+ * __rollback_evict_exclusive_apply --
+ *     Apply an eviction helper to all dhandles.
+ */
+static int
+__rollback_evict_exclusive_apply(WT_SESSION_IMPL *session, int (*dhandle_func)(WT_SESSION_IMPL *))
+{
+    WT_CURSOR *metadata_cursor;
+    WT_DECL_RET;
+    const char *uri;
+
+    WT_RET(__wt_metadata_cursor(session, &metadata_cursor));
+
+    while ((ret = metadata_cursor->next(metadata_cursor)) == 0) {
```

Comment:
@kommiharibabu had an idea to block eviction at the connection level instead of doing this to avoid expensive iteration over metadata.

That would be more performant but adds another mechanism for controlling eviction instead of using the primitives we already have.